application/models/user/user.php

```diff
@@ -0,0 +1,375 @@
+<?php
+
+namespace User;
+
+require_once './vendor/autoload.php';
+use Database\Database;
+
+/**
+ *
+ * @method Signup_User
+ *
+ */
+class User extends Database{
+
+
+  private $username;
+  private $password;
+  private $email;
+  private $gender;
+  private $bio;
+
+  /**
+   *
+   * @param mixed $email
+   * @param mixed $username
+   * @param mixed $password
+   * @param mixed gender
+   * @param mixed bio
+   *
+   * @return boolean
+   *
+   */
+
+  public function Signup_User($username, $email, $password, $gender, $bio) {
+
+    $this->username = $username;
+    $this->$password = $password;
+    $this->email = $email;
+    $this->gender = $gender;
+    $this->bio = $bio;
+
+    $sql="SELECT * FROM Users WHERE Email='".$email."'";
+
+    $check= $this->link->query($sql);
+    $row_count=$check->num_rows;
+
+    if($row_count == 0) {
+      $sql1="INSERT INTO Users (Username,Password,Email,Gender,Bio) VALUES ('".$username."','".md5($password)."','".$email."','".$gender."','".$bio."')";
+      $result=mysqli_query($this->link,$sql1);
+      return $result;
+    }
+
+    else{
+      return false;
+    }
+
+}
+
+    /**
+     * This function is used to validate the login process
+     * @param mixed $username
+     * @param mixed $password
+     *
+     * @return boolean
+     *
+     */
+
+  public function validate_login($username,$password) {
+
+    $this->username = $username;
+    $this->$password = $password;
+
+    // The md5 functions is being used to encrypt the password inside database
+    $sql="SELECT Uid from Users WHERE Email='".$username."' and Password='".md5($password)."'";
+    $output=mysqli_query($this->link,$sql);
+    $data=mysqli_fetch_array($output);
+    $row_count=$output->num_rows;
+
+    if($row_count==1){
+      $_SESSION['logged_in']=true;
+      $_SESSION['User_id']=$data['Uid'];
+      return true;
+    }
+    else {
+      return false;
+    }
+
+}
+
+
+  /*
+    Regular Expression: $\S*(?=\S{8,})(?=\S*[a-z])(?=\S*[A-Z])(?=\S*[\d])(?=\S*[\W])\S*$
+    $ = beginning of string
+    \S* = any set of characters
+    (?=\S{8,}) = of at least length 8
+    (?=\S*[a-z]) = containing at least one lowercase letter
+    (?=\S*[A-Z]) = and at least one uppercase letter
+    (?=\S*[\d]) = and at least one number
+    (?=\S*[\W]) = and at least a special character (non-word characters)
+    $ = end of the string
+ */
+  /**
+   * This function will be used to validate the password entered by user
+   *
+   * @param mixed $password
+   *
+   * @return boolean
+   *
+   */
+
+  public function Validate_Password($password){
+
+    if (!preg_match_all('$\S*(?=\S{8,})(?=\S*[a-z])(?=\S*[A-Z])(?=\S*[\d])(?=\S*[\W])\S*$', $password)) {
+      return FALSE;
+    }
+    else {
+      return TRUE;
+    }
+
+  }
+
+  /**
+   *
+   * This function is being used to fetch data from the field
+   *
+   * @param mixed $email
+   *
+   * @return string
+   *
+   */
+  public function Get_Name($email){
```

Comment:
Please add space before `{` brace, also do same changes below of this where required.